client.py

```diff
@@ -156,6 +174,9 @@ def main():
 
                 if "count" in s and s["count"] > limit:
                     s["count"] = limit
+                if ("max_memory_per_job" in config_cluster 
+                    and s["memory"] > config_cluster["max_memory_per_job"]):
+                    continue
```

Comment:
These three lines aren't needed anymore. They are covered by the for loop above.